digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/process/CommonHistoryFiller.scala

```diff
@@ -0,0 +1,157 @@
+package fi.liikennevirasto.viite.process
+
+
+import fi.liikennevirasto.digiroad2.dao.Sequences
+import fi.liikennevirasto.viite.NewRoadAddress
+import fi.liikennevirasto.viite.dao.{LinkStatus, ProjectLink, RoadAddress}
+
+object CommonHistoryFiller {
+  private def applyUnchanged(currentRoadAddresses : Seq[RoadAddress])(projectLinks: Seq[ProjectLink], newRoadAddresses: Seq[RoadAddress]): Seq[RoadAddress] = {
+    val unchangedLinks = projectLinks.filter(_.status == LinkStatus.UnChanged)

```

Comment:
We could use a partition instead of `fitler` here and then in the end `filterNot`

Reply:
Here we are filtering project_links, and in the end we are filtering road_addresses